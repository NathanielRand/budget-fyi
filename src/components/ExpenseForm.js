import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

const date = new Date();
const now = moment();
console.log(date);
console.log(now.format("dddd, MMMM Do YYYY, h:mm:ss a"));

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: props.expense ? props.expense.description : "",
      note: props.expense ? props.expense.note : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocused: false,
      error: undefined
    };
  }
  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description: description }));
  };
  onNoteChange = e => {
    const note = e.target.value;
    this.setState(() => ({ note: note }));
  };
  onAmountChange = e => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount: amount }));
    }
  };
  onDateChange = createdAt => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  onSubmit = e => {
    e.preventDefault();

    if (!this.state.description || !this.state.amount) {
      // Set error state equal to "Please provide description and amount."
      this.setState(() => ({
        error: "Please provide description and amount"
      }));
    } else {
      // Clear the error
      this.setState(() => ({
        error: ""
      }));
      // onSubmit
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };
  render() {
    return (
      <div className="form__background">
        <form className="form" onSubmit={this.onSubmit}>
          {this.state.error && (
            <p className="form__error">{this.state.error}</p>
          )}
          <input
            type="text"
            placeholder="Description"
            autoFocus
            className="text-input"
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            type="text"
            placeholder="Amount"
            className="text-input"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <SingleDatePicker
            className="date-input"
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <label>
            <input
              className="form-check-input"
              type="radio"
              name="react-tips"
              value="option1"
              checked={true}
            />
            Occurance
          </label>
          <textarea
            placeholder="Add an expense note (optional)"
            className="textarea"
            value={this.state.note}
            onChange={this.onNoteChange}
          ></textarea>
          <div className="form__actions">
            <button className="form__button">Save Expense</button>
          </div>
        </form>
      </div>
    );
  }
}
