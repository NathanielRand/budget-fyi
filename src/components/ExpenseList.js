import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectExpense from "../selectors/expenses";

export const ExpenseList = props => (
  <div>
    <p>$1000 (EXPENSES) / $1300 (INCOME)</p>
    <p>Expense Total: </p>
    <p>Expense Count: {props.expenses.length}</p>
    <p>Expense Average: (length divided by total then round up to floor)</p>
    {props.expenses.length === 0 ? (
      <p>No Expenses</p>
    ) : (
      props.expenses.map(expense => {
        return <ExpenseListItem key={expense.id} {...expense} />;
      })
    )}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: selectExpense(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
