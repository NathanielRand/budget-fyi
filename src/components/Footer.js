import React from "react";
import { Link } from "react-router-dom";

function Footer() {
  return (
    <div className="footer">
      <div className="footer-title">
        <Link to="/">
          <h3>BudgetFYIâ„¢</h3>
        </Link>
      </div>
      <div className="footer-content">
        <Link to="/about">
          <h5>About</h5>
        </Link>
        <Link to="/help">
          <h5>Help</h5>
        </Link>
        <Link to="/contact">
          <h5>Contact</h5>
        </Link>
      </div>
    </div>
  );
}

export default Footer;
