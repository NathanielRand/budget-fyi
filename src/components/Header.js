import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link to="/dashboard">
          <h1 className="header__title">
            Budget<b className="header__title-fyi">FYI</b>
          </h1>
        </Link>
        <button className="header__button" onClick={startLogout}>
          LOGOUT
        </button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
