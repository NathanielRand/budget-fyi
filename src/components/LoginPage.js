import React from "react";

export const LoginPage = () => (
  <div>
    <h1>Welcome to BudgetFYI</h1>
    <button>LOGIN</button>
  </div>
);
