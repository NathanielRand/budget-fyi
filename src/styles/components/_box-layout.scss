.box-layout {
  background: url("/images/Shore2.jpg");
  background-size: cover;
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.box-layout__box {
  background: #546e7a;
  padding: 5%;
  border: 2px white solid;
  border-radius: 22px 66px;
  width: 45rem;
  text-align: center;
  color: white;
  @media (max-width: $desktop-breakpoint) {
    margin-top: $l-size;
    padding-left: $s-size;
    width: 35rem;
  }
}

.box-layout__title {
  font-size: $font-size-x-lrg;
}

.box-layout__sub-title {
  font-size: $font-size-med;
  font-style: italic;
}

.box-layout__button {
  background: $primary-gradient; /* fallback for old browsers */
  background: -webkit-linear-gradient(
    $primary-gradient
  ); /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(
    $primary-gradient
  ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  border: 2px white solid;
  border-radius: 11px 33px;
  color: #37474f;
  font-family: "Rajdhani";
  font-size: $font-size-med;
  margin-top: 4rem;
  margin-bottom: 3rem;
  padding: 2.2rem;
}

.landing-title-fyi {
  // Reference: https://uigradients.com/#SublimeLight
  background: #70e1f5; /* fallback for old browsers */
  background: -webkit-linear-gradient(
    to top,
    #70e1f5,
    #ffd194
  ); /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(
    to top,
    #70e1f5,
    #ffd194
  ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}
