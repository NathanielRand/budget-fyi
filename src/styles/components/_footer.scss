@import url("https://fonts.googleapis.com/css?family=Rajdhani:400,600,700&display=swap");

.footer {
  background: #546e7a;
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
  border-top: 4px #eceff1 solid;
  font-family: "Rajdhani";
}

.footer-title {
  background: #70e1f5; /* fallback for old browsers */
  background: -webkit-linear-gradient(
    to top,
    #70e1f5,
    #ffd194
  ); /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(
    to top,
    #70e1f5,
    #ffd194
  ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  font-size: $font-size-med;
  margin-left: 3rem;
  @media (min-width: $desktop-breakpoint) {
    margin-left: 30%;
    font-size: $font-size-lrg;
  }
}

.footer-content {
  color: #cfd8dc;
  font-family: "Rajdhani";
  font-size: $font-size-med;
  margin-right: 6rem;
  text-decoration: none !important;
  background: #70e1f5; /* fallback for old browsers */
  background: -webkit-linear-gradient(
    to top,
    #70e1f5,
    #ffd194
  ); /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(
    to top,
    #70e1f5,
    #ffd194
  ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  @media (min-width: $desktop-breakpoint) {
    margin-right: 40%;
  }
}
