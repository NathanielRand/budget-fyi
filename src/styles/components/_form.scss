.form {
  display: flex;
  flex-direction: column;
  padding-top: $xl-size;
  > * {
    margin-bottom: $m-size;
  }
}

.form__background {
  background: #607d8b;
}

.form__error {
  margin: 0 0 $m-size 0;
  font-style: italic;
  text-align: center;
}

.form__actions {
  text-align: center;
}

.form__button {
  background: $primary-gradient; /* fallback for old browsers */
  background: -webkit-linear-gradient(
    $primary-gradient
  ); /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(
    $primary-gradient
  ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  border: 2px white solid;
  border-radius: 9px 27px;
  color: #37474f;
  font-family: "Rajdhani";
  font-size: $font-size-med;
  margin-top: 1rem;
  margin-bottom: 3rem;
  padding: 2rem;
}
