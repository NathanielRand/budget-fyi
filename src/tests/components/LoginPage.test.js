import React from "react";
import { shallow } from "enzyme";
import { LoginPage } from "../../components/LoginPage";

test("Should render the Login Page", () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test("Should call startLogin on button click", () => {
  // Create the spy
  const startLogin = jest.fn();
  // Pass the spy into the login page
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  // Click the button
  wrapper.find("button").simulate("click");
  // Expect spy to have been called
  expect(startLogin).toHaveBeenCalled();
});
