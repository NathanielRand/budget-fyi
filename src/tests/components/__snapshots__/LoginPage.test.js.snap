// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Login Page 1`] = `
<div>
  <h1>
    Welcome to BudgetFYI
  </h1>
  <button>
    LOGIN
  </button>
</div>
`;
